src/fidesops/ops/schemas/manual_webhook_schemas.py

```diff
@@ -0,0 +1,67 @@
+from typing import List, Optional, Set
+
+from fideslib.utils.text import to_snake_case
+from pydantic import ConstrainedStr, validator
+
+from fidesops.ops.schemas.base_class import BaseSchema
+from fidesops.ops.schemas.connection_configuration.connection_config import (
+    ConnectionConfigurationResponse,
+)
+
+
+class ManualWebhookFieldType(ConstrainedStr):
+    """Using ConstrainedStr instead of constr to keep mypy happy"""
+
+    min_length = 1
+    max_length = 200
+
+
+class ManualWebhookField(BaseSchema):
+    """Schema to describe the attributes on a manual webhook field"""
+
+    pii_field: ManualWebhookFieldType
+    dsr_package_label: Optional[ManualWebhookFieldType]
+
+    class Config:
+        orm_mode = True
+
+
+class AccessManualWebhooks(BaseSchema):
+    """Expected request body for creating Access Manual Webhooks"""
+
+    fields: List[ManualWebhookField]
+
+    class Config:
+        orm_mode = True
+
+    @validator("fields")
+    def check_for_duplicates(
+        cls, value: List[ManualWebhookField]
+    ) -> List[ManualWebhookField]:
+        """
+        Verify that pii_fields and dsr_package_labels are unique.
+
+        Set the dsr_package_label to a snake_cased lower case version of pii field if it doesn't exist.
+        """
+        unique_pii_fields: Set[str] = {field.pii_field for field in value}
+        if len(value) != len(unique_pii_fields):
+            raise ValueError("pii_fields must be unique")
+
+        for field in value:
+            if not field.dsr_package_label:
+                field.dsr_package_label = to_snake_case(field.pii_field)
+
+        unique_dsr_package_labels: Set[Optional[str]] = {
+            field.dsr_package_label for field in value
+        }
+        if len(value) != len(unique_dsr_package_labels):
+            raise ValueError("dsr_package_labels must be unique")
```

Comment:
Would it work to flip these so you could bail early if the lengths don't match or does it need to be converted to snake case first?

```suggestion
        unique_dsr_package_labels: Set[Optional[str]] = {
            field.dsr_package_label for field in value
        }
        if len(value) != len(unique_dsr_package_labels):
            raise ValueError("dsr_package_labels must be unique")

        for field in value:
            if not field.dsr_package_label:
                field.dsr_package_label = to_snake_case(field.pii_field)
```

Reply:
It's a version of the pii_field specifically that I'm converting to snake case, so I wanted to do the conversion first in case the conversion collided with an already existing dsr_package_label!